src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -180,48 +186,56 @@ public Response intercept(Chain chain) throws IOException {
 	/** 
 	 * If we have a refreshable token and it is expired or near expiry, do a token refresh 
 	 * @param chain - The intercepted chain
-	 * @param baseUrl - The base hose url from the request, used to issue the refresh request
+	 * @param zone5Url - The base hose url from the request, used to issue the refresh request
 	 **/
-	private void refreshIfRequired(Chain chain, String baseUrl) {
+	//auto refresh needs to use this configured baseURL for constructing the auto refresh request,
+	// not the baseURL of the current incoming request.
+	private void refreshIfRequired(Chain chain, String zone5Url) {
+		RequestBody body = null;
 		// check the token for expiry
 		// note that refresh itself does not require auth so will not end up cyclicly back here
-		AuthToken token = this.token.get();
-		if (token != null && token.getRefreshToken() != null && token.isExpired() && chain != null && baseUrl != null && !baseUrl.isEmpty()) {
+		AuthToken token = this.clientConfig.getToken();
+		if (token != null && token.getRefreshToken() != null && token.isExpired() && chain != null && zone5Url != null && !zone5Url.isEmpty()) {
 			// because requests can be concurrent, we need to synchronize so that we only do one refresh for an expired token
 			// and all of the requests dependent on that refresh are queued until the token is refreshed
 			synchronized(fetchTokenLock) {
 				// once we are inside the mutex block we need to re-test our token because it might have
 				// been refreshed while we were waiting
-				token = this.token.get();
+				token = this.clientConfig.getToken();
 				if (token != null && token.isExpired()) {
 					try {
-						if (token.getRefreshToken() != null) {
-							// do a Cognito refresh
-							String username = token.extractUsername();
-							if (username != null) {
-								RequestBody body = new FormBody.Builder().add("client_id", clientID)
-																		 .add("client_secret", clientSecret)
-																		 .add("grant_type", GrantType.REFRESH_TOKEN.toString())
-																		 .add("username", username)
-																		 .add("refresh_token", token.getRefreshToken()).build();
-								String url = baseUrl + Users.NEW_ACCESS_TOKEN;
-								Request authRequest = new Request.Builder().url(url).post(body).build();
```

Comment:
Note: Before when this was using the getBaseUrl we were guaranteed that this was formatted correctly with "https://" prefix. Now we are accepting zone5BaseUrl as a String and it may not be properly formatted. We probably need to either do our own sanity checking on the passed in string, OR we could make the passed in zone5BaseUrl of type URL so that we know it is properly constructed. Then we can just append Users.NEW_ACCESS_TOKEN to the URL. This line `Request authRequest = new Request.Builder().url(url).post(body).build();` works with a String url or a URL url so we would not need to convert it to a String.